### ISSUE TITLE
Noobie Question and issue

### ISSUE THREAD

**Issue by robboz4** [sarnau/EnphaseEnergy#3]
I'm trying to install the plugin and it did work but nothing shows up the homebridge display. I don't understand the procedure to add it to the HomeKit app on the iPhone. I already have it loaded and working with WEMO, HUE accessories. I reread the instructions and added the generated password to the installer  field   and now I get this error in the log:
 [enphaseEnvoy] Device: 192.168.1.8 Envoy-S, requesting envoyBackboneAppData error: Error: connect EHOSTUNREACH 192.168.1.8:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16) {
  errno: -113,
  code: 'EHOSTUNREACH',
  syscall: 'connect',
  address: '192.168.1.8',
  port: 80,
  config: [Object],
  request: [Writable],
  response: undefined,
  isAxiosError: true,
  toJSON: [Function: toJSON]

I'm running Homebridge on a Pi:

 Linux Homebridge 5.10.60-v7+ #1449 SMP Wed Aug 25 15:00:01 BST 2021 
armv7l GNU/Linux
Known as: Homebridge
OS Version : Raspbian GNU/Linux 10 (buster)

 Here is the Envoy-S  config info

"devices": [
                {
                    "name": "Envoy-S",
                    "host": "192.168.1.8",
                    "refreshInterval": 60,
                    "disableLogInfo": false,
                    "installerPasswd": "6e9f3c2",
                    "powerProductionMaxDetected": 100,
                    "energyProductionLifetimeOffset": 0,
                    "powerConsumptionTotalMaxDetected": 1000,
                    "energyConsumptionTotalLifetimeOffset": 0,
                    "powerConsumptionNetMaxDetected": 0,
                    "energyConsumptionNetLifetimeOffset": 0
                }
            ],
            "platform": "enphaseEnvoy"

There are no errors with the JSON config file. This is a brand new Enphase Solar installation. I have 15 panels and a battery system  installed. The OS software Nodejs and Homebridge are all the latest version

So any assistance would be appreciated to get me up and running.

Cheers

Dave

**Comment by dlmcpaul** [comment]
Enphase have changed the way local access is provided with the latest firmware.  See this thread https://community.enphase.com/s/question/0D53m00006ySLuR/unimpressed-with-loss-of-local-api-connectivity-to-envoy

If your firmware is **higher** than 5.0.55 then I suspect this program and others will not work until a fix is available.

So first step is to check what firmware you are running.

Other than that, the error suggest there is no connectivity between your Pi and the envoy.  If you remote into your pi can you ping 192.168.1.8?

**Comment by sarnau** [comment]
I have 5.0.55 and all scripts are working fine. They are based on Enphase Android application, which they probably did not want to break.

**Comment by robboz4** [comment]
It looks like I have D6.0.98 on the IQ Envoy. 
Also  the IP address stopped responding to pings, so I guess that was the connect error. No idea why it stopped responding. I will have see what today's IP address! I also haven't had much look trying to get access to the web page that is hosted on the Envoy either. It timeout most times or doesn't respond. I can only use the iPhone app. May be I'll contact Envoy support and ask them about that.  

Thanks for the info. I've removed the plugin for now.

Cheers

**Comment by eleventhaus** [comment]
I'm on software version D7.0.71 and this script isn't working. Guessing they updated the firmware to prevent owners from accessing their own systems. How is this legal?

I should be able to access my own property and do whatever I want with it.

**Comment by sarnau** [comment]
Yeah, "your own property". I know, only because we _paid_ for it, being able to access to it… Would be nice to sue them about this, because this shitty behavior.

**Comment by sarnau** [comment]
It seems the hardware never updates, so my Envoy is still at D5.0.62.

**Comment by dlmcpaul** [comment]
Since getting my envoy I have seen 3 updates applied.  I think it depends on your installer who have the password to apply updates.

**Comment by sarnau** [comment]
My installer seems to be "dead"…

**Comment by dgarciam** [comment]
So, this means we have to wait for someone else to crack the envoy again to actually include this on home assistant?

**Comment by Matthew1471** [comment]
@dgarciam no, just requires you to get a token from their website that expires every 6 months that you use as the password.

If you get a shorter timeframe like 1 or 12 hours then your token was generated by someone signing in as an installer and not as the system owner.

The token can even be used offline.. it just has a 6 month death date in which you'll then need to go get another.
